examples/gno.land/p/demo/accesscontrol/accesscontrol.gno

```diff
@@ -0,0 +1,185 @@
+package accesscontrol
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ownable"
+)
+
+const (
+	RoleName     = "roleName"
+	Sender       = "sender"
+	Account      = "account"
+	NewAdminRole = "newAdminRole"
+)
+
+// Role struct to store role information
+type Role struct {
+	Name    string
+	Holders *avl.Tree // std.Address -> struct{}
+	Ownable *ownable.Ownable
+}
+
+// Roles struct to store all Roles information
+type Roles struct {
+	AllRoles []*Role
+	Ownable  *ownable.Ownable
+}
+
+func validRoleName(name string) error {
+	if len(name) > 30 || name == "" {
```

Comment:
the limit of 30 is intended to prevent someone from too overloading the chaine, for example by setting it to 1M characters. Do you think it's better to set a higher maximum or to let the user do it himself, even if this leaves open the possibility of defining a large number of characters?